cores/esp32/esp32-hal-time.c

```diff
@@ -45,6 +45,11 @@ static void setTimeZone(long offset, int daylight)
  * */
 void configTime(long gmtOffset_sec, int daylightOffset_sec, const char* server1, const char* server2, const char* server3)
 {
+#ifndef _TCPIP_ADAPTER_H_
```

Comment:
This doesn't seem to be sufficient, it just checks that tcpip_adapter.h has been included. Perhaps there should be a check against ```WiFi.getMode() != WIFI_MODE_NULL)``` or some variant there of via a generic function call (in core vs in wifi lib)